W-10673246 adding content for running mutations (#70)
mulesoft/docs-anypoint-datagraph: modules/ROOT/pages/query-unified-schema.adoc

= Running Queries and Mutations

The unified schema resolves into a single GraphQL endpoint. To run queries or mutations against the unified schema, you first request access (if applicable) and then write and run a GraphQL query in the query editor.

> == Prerequisites

Reviewer: == Before You Begin

> == Prerequisites

Author: ```suggestion
== Before You Begin
```

> * You must have xref:permissions.adoc[the Consume, Contribute, or Admin permission] to use request access and run queries and mutations.

Reviewer: ```suggestion
* You must have xref:permissions.adoc[the Consume, Contribute, or Admin permission]. 
```
We don't _have_ to tell _why_ in a prereq. list item. Your choice. If you keep...:
```suggestion
* You must have xref:permissions.adoc[the Consume, Contribute, or Admin permission] to run queries and mutations.
```

. Click *Run*.
+
image::datagraph-qsg-query-result.png[Query editor displays query results]
. To see results with query tracing, from the actions menu (*...*), select *Trace Query*, and then click *Run*.
+
image::datagraph-qsg-query-traces.png[Query trace view open with query results]
+
Query trace results for DataGraph provide the following information:
+
* Time taken by DataGraph to parse and validate the query
* Total response time for the entire query
* Duration of requests to each source API in the query

. To see response logs associated with the query, from the actions menu (*...*), select *View Response Logs*.
+
image::datagraph-qsg-response-logs.png[Query response logs page]
+
Log levels for DataGraph include DEBUG, INFO, WARN, and ERROR.
. Click *View History* to access this same query (or others) later.
. To use this query in your application, click *Copy endpoint* to copy the query and the automatically generated GraphQL endpoint. You can also copy the query as a curl snippet.

== Run a Mutation

> Use the query editor to write, run, and troubleshoot mutations.

Author: ```suggestion
Use the query editor to write and then run mutations. If needed, you can also troubleshoot mutations in the query editor
```

. Click *Run*.
+
image::datagraph-qsg-query-result.png[Query editor displays query results]
. To see results with query tracing, from the actions menu (*...*), select *Trace Query*, and then click *Run*.
+
image::datagraph-qsg-query-traces.png[Query trace view open with query results]
+
Query Trace results for DataGraph provide the following information:
+
* Time taken by DataGraph to parse and validate the query
* Total response time for the entire query
* Duration of requests to each source API in the query

. To see response logs associated with the query, from the actions menu (*...*), select *View Response Logs*.
+
image::datagraph-qsg-response-logs.png[Query response logs page]
+
Log levels for  DataGraph include, DEBUG, INFO, WARN, and ERROR.

> . Click *View History* to access this same query—or others—at a later time.

Reviewer: ```suggestion
. Click *View History* to access this same query (or others) later.
```

. To use this query in your application, click *Copy endpoint* to copy the query and the automatically generated GraphQL endpoint. You can also copy the query as a cURL snippet.

== Run a Mutation

Use the query editor to write, run, and troubleshoot mutations.

> . If any of the APIs in the unified schema require pass-through authentication, click *Add HTTP Headers*, and:

Reviewer: ```suggestion
. If any of the APIs in the unified schema require pass-through authentication, click *Add HTTP Headers* and:
```

.. Select a header name.
.. Enter a header value.

> .. To specify additional header name and value pairs for other APIs, click *+Add Header* x 

Reviewer: x?
```suggestion
.. To specify additional header name and value pairs for other APIs, click *+Add Header*.
```

.. Click *Save*.
.. To edit the HTTP headers, click *Edit HTTP Headers*.

> . In the query editor, start typing your mutation using the in-line autocomplete function or type *Option/Alt + Shift + Space* to open the autocomplete menu.

Reviewer: ```suggestion
. In the query editor, start typing your mutation using the in-line autocomplete function, or type *Option/Alt + Shift + Space* to open the autocomplete menu.
```

+
image::datagraph-mutation-autocomplete.png[Using inline autocompletion to write a mutation,70%]

. Optionally, in the **Query Variables** pane, add any variables you want to use.

> . Click **Run**. 

Reviewer: Double *s okay?

> . Click **Run**. 

Author: ```suggestion
. Click *Run*. 
```

+
image::datagraph-mutation-result.png[Query editor displays mutation results]

> . To see results with query tracing, from the actions menu (*...*), select *Trace Query*, and then click *Run*.

Reviewer: ```suggestion
. To see results with query tracing, from the actions menu (*...*), select *Trace Query* and then click *Run*.
```

. To see response logs associated with the query, from the actions menu (*...*), select *View Response Logs*.

> . Click *View History* to access this same mutation—or others—at a later time.

Reviewer: ```suggestion
. Click *View History* to access this same mutation (or other) later.
```
